File: Data Structures & Algorithms/three-integer-sum/submission-0.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        vector<vector<int>> ans;

        std::sort(nums.begin(), nums.end());
        unordered_map<int, int> key;
        for (int i = 0; i < nums.size();++i){
            key[nums[i]]++;
        }

        for (int i = 0; i < nums.size(); i++) {
            key[nums[i]]--;
            if (i > 0 && nums[i] == nums[i - 1]) continue;

            for (int j = i + 1; j < nums.size(); j++) {
                key[nums[j]]--;
                if (j > i + 1 && nums[j] == nums[j - 1]) continue;

                int target = -(nums[i] + nums[j]);
                if (key[target] > 0) {
                    ans.push_back({nums[i], nums[j], target});
                }
            }

            for (int j = i + 1; j < nums.size(); j++) {
                key[nums[j]]++;
            }
        }

        return ans;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer, Integer> map = new HashMap<>();

        for (int i : nums){
            int old = map.getOrDefault(i, 0);
            map.put(i, old + 1);
        }

        ArrayList<int[]> arr = new ArrayList<>();
        for (Map.Entry<Integer, Integer> entry : map.entrySet()){
            arr.add(new int[] {entry.getValue(), entry.getKey()});
        }

        arr.sort((a,b) -> b[0] - a[0]);

        int[] ans = new int[k];
        for (int i = 0; i < k; i++){
            ans[i] = arr.get(i)[1];
        }
        return ans;
    }
}
